symfony/plugins/orangehrmLeavePlugin/modules/leave/actions/baseLeaveAction.php

```diff
@@ -28,7 +28,7 @@ protected function setLeaveTypeService(LeaveTypeService $service) {
      * @param type $dataGroups
      * @return type
      */
-    public function getDataGroupPermissions($dataGroups, $self = false) {
+    protected function getDataGroupPermissions($dataGroups, $self = false) {
```

Comment:
@RajithaKumara How method access cause to that issue? Still, It allows to overridden via subclasses

Reply:
Yeah. I commented error stack under JIRA. The error says cannot override public method by protected function at the child class. Otherwise, it will be solve child classes method 'protected function getDataGroupPermissions()' change as 'public function getDataGroupPermissions()'. Is it violate OOP?

Reply:
@samantha: Checked and this is due to Rajitha extending all Leave Action classes from the  baseLeaveAction. in BaseLeaveAction the getDataGroupPermissions() is marked public, but in some Action classes that previously did NOT extend from the baseAction had the getDataGroupPermissions() method marked 'protected'. The Rule with overriding is that the overridden method can have the same or more relaxed access modifiers than the base function. Here the derived class has more restrictive access, which threw the error. Marking the base method as protected will resolve the issue.